gcp/README.md

```diff
@@ -324,48 +324,73 @@ See [Getting started: One-time Stackdriver Workspace setup](README.md#one-time-s
+1. Delete associated PVC and then affected pod. For our example case:
    * `kubectl --namespace gpii delete pvc database-storage-couchdb-couchdb-1`
    * `kubectl --namespace gpii delete pod couchdb-couchdb-1`
-* After target pod is terminated, Persistent Disk that was mounted into it thru corresponding PVC will be destroyed as well.
-* When new pod is created to replace deleted one, corresponding PVC will be created as well, and, thru it, new PV object for new GCE PD.
-* Run `rake deploy_module[couchdb]` to patch newly created PV with annotations for `k8s-snapshots`.
-* CouchDB cluster will replicate data to recreated node automatically.
-* Corrupted node is now recovered.
+1. After target pod is terminated, Persistent Disk that was mounted into it thru corresponding PVC will be destroyed as well.
+1. When new pod is created to replace deleted one, corresponding PVC will be created as well, and, thru it, new PV object for new GCE PD.
+1. Run `rake deploy_module[couchdb]` to patch newly created PV with annotations for `k8s-snapshots`.
+1. CouchDB cluster will replicate data to recreated node automatically.
+1. Corrupted node is now recovered.
    * You can check DB status on recovered node with `kubectl exec --namespace gpii -it couchdb-couchdb-N -c couchdb -- curl -s http://$TF_VAR_couchdb_admin_username:$TF_VAR_couchdb_admin_password@127.0.0.1:5984/gpii/`, where N is node index.
 
+### The CouchDB cluster won't converge because one of its disks is in the wrong zone
+
+Consider this scenario:
+* A Kubernetes cluster has 3 Nodes, one each in us-central1-a, us-central1-b, and us-central1-c.
+* On this Kubernetes cluster runs a CouchDB cluster with 3 replicas, one on each Node (and thus one in each Zone).
+* Each CouchDB replica is attached to a Persistent Volume, which is tied to a specific Zone.
+* A change is deployed which requires the Kubernetes cluster to be destroyed and re-created. The new Kubernetes cluster has 3 Nodes, but now they are in us-central1-a, us-central1-b, and us-central1-f (i.e. no Nodes in us-central1-c).
+* CouchDB is deployed. The replicas in us-central1-a and us-central1-b are scheduled so that they can re-attach to the Persistent Volumes containing their data.
+* But the replica that wants to re-attach to the PV in us-central1-c cannot do so, because there is no Node in us-central1-c on which to run.
+   * You'll know you're in this state if one CouchDB Pod can't start at all (`0/2 running`), and there are messages about `unschedulable` due to `Persistent Volume` in the output of `rake display_cluster_state`.
+
+The workaround is to manually move the Persistent Disk that backs the Persistent Volume into a Zone where it can be attached by a CouchDB replica:
+1. Find a Zone that has a Node with no CouchDB Pod running on it. Note this for later.
+1. Open Google Cloud console, go to "Compute Engine" -> "Disks".
```

Comment:
do you think that we can use the command [gcloud compute disks move](https://cloud.google.com/sdk/gcloud/reference/compute/disks/move) instead of the following 4 steps?

Reply:
Initially, I couldn't get this to work. It seems that the PV (or PVC) expects to find the underlying Disk in a specific Zone:

`AttachVolume.Attach failed for volume "pvc-344f6c0e-4074-11e9-a476-42010a80015b" : GCE persistent disk not found: diskName="gke-k8s-cluster-010227-pvc-344f6c0e-4074-11e9-a476-42010a80015b" zone="us-central1-c"`

However, we can change the expected Zone with `kubectl edit`.

I kept both approaches documented, just in case, but I agree that using `gcloud` to move disks is easier than doing a snapshot-restore cycle.